pkg/nodeconfig/nodeconfig.go

```diff
@@ -227,13 +227,13 @@ func (nc *nodeConfig) Configure() error {
 
 		// Configuring cloud node manager for Azure platform with CCM support
 		if ownedByCCM && nc.platformType == configv1.AzurePlatformType {
```

Comment:
Should this taint removal `if` block be moved after the `nc.configureWICD()`? This will ensure the taint is removed only after the CCM service has successfully started.